certification/internal/policy/types.go

```diff
@@ -1,38 +1,27 @@
 package policy
 
-type Definition struct {
-	ValidatorFunc func(string) (bool, error)
-	Metadata      Metadata
-	HelpText      HelpText
-}
-
-func (pd *Definition) Validate(image string) (bool, error) {
-	return pd.ValidatorFunc(image)
-}
-
-func (pd *Definition) Meta() Metadata {
-	return pd.Metadata
-}
-
-func (pd *Definition) Help() HelpText {
-	return pd.HelpText
-}
-
-type PolicyInfo struct {
-	Metadata `json:"metadata" xml:"metadata"`
-	HelpText `json:"helptext"`
+// Policy as an interface containing all methods necessary
+// to use and identify a given policy.
+type Policy interface {
+	// Validate whether the asset enforces the policy.
+	Validate(image string) (result bool, err error)
+	// return the name of the policy
+	GetName() string
+	// return the policy's metadata
+	GetMetadata() Metadata
+	// return the policy's help text
+	GetHelp() HelpText
```

Comment:
Defining the policy here means that it is no longer an exported API. This needs to live in certification.go, or somewhere outside of the `internal/` subdirectory.

Reply:
A top-level policy package would be where I would put it.